### PR TITLE
New Frontend Boilerplate/Setup - It has begun

### PR DESCRIPTION
**Pull request by aryanshah701**
# Description

- A new NextJS frontend with the common package setup called frontend-v2
- Update CI to run linting on this directory 
- Update dev:v2 script to run this new frontend with the api-v2

Closes # [Setup NextJS Frontend](https://trello.com/c/P5jLQRxT/402-setup-nextjs-for-frontend)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update
- [x] This requires a run of `yarn install`

# How Has This Been Tested?

- Ensure db is running with yarn dev:db:up and then run yarn dev:v2 to run the new frontend with api-v2. You should be able to see a NextJS app at http://localhost:3002/  and also see that imports from common package work.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -3,6 +3,17 @@ name: CI
 on: [push]
 
 jobs:
+  run-frontend-v2-lint:
+    runs-on: ubuntu-latest
+    name: Run linting for frontend v2
+    steps:
+      - uses: actions/checkout@v2
```
Comment (bugsalexander) [inline .github/workflows/main.yml:10]: use v3

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -3,6 +3,17 @@ name: CI
...
+      - name: install node v12
+        uses: actions/setup-node@v1
```
Comment (bugsalexander) [inline .github/workflows/main.yml:12]: use v3

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -3,6 +3,17 @@ name: CI
...
+        with:
+          node-version: 12
```
Comment (bugsalexander) [inline .github/workflows/main.yml:14]: do we wanna use v16?

File: README.md [hunk 4]
```diff
@@ -1,5 +1,15 @@
 # GraduateNU
 
+## New Version
```
Comment (bugsalexander) [inline README.md:3]: refer to as v2

File: packages/frontend-v2/.eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  extends: "next/core-web-vitals",
```
Comment (bugsalexander) [inline packages/frontend-v2/.eslintrc.js:2]: is there a way to make it extend the workspace root eslint config as well?

File: packages/frontend-v2/package.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@graduate/common": "workspace:^",
```
Comment (bugsalexander) [inline packages/frontend-v2/package.json:12]: use `workspace:*`, see https://yarnpkg.com/features/workspaces/#workspace-ranges-workspace

File: packages/frontend-v2/package.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    "next": "12.1.4",
+    "react": "18.0.0",
```
Comment (bugsalexander) [inline packages/frontend-v2/package.json:14]: pogg

File: packages/frontend-v2/styles/globals.css [hunk 8]
```diff
@@ -0,0 +1,16 @@
+html,
```
Comment (bugsalexander) [inline packages/frontend-v2/styles/globals.css:1]: delete?

File: packages/frontend/Dockerfile [hunk 9]
```diff
@@ -10,6 +10,7 @@ COPY babel.config.js .
 COPY package.json .
 COPY packages/api/package.json packages/api/package.json
 COPY packages/frontend/package.json packages/frontend/package.json
+COPY packages/frontend-v2/package.json packages/frontend-v2/package.json
```
Comment (bugsalexander) [inline packages/frontend/Dockerfile:13]: we should probably figure out a way to just copy recursively all package.jsons listed in the workspace... 

[this blog post](https://www.huy.dev/yarn-v2-workspace-docker-vs-code-2020-03-23/) does this via [this weird command](https://github.com/huy-nguyen/yarn-v2-workspace-docker-vs-code/blob/6c427bf97a4a17ddc4bbb911b6c473c55d9667a3/Dockerfile#L18) – though feel free to just kick the can for later

**Review by bugsalexander — COMMENTED** [review]
a few small fixes

File: packages/frontend-v2/README.md [hunk 10]
```diff
@@ -0,0 +1,18 @@
+# GraduateNU Frontend V2
+
+This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
+
+## Getting Started
+
+Run the application(both frontend and backend) with `yarn dev` from the root of the monorepo.
```
Comment (cindy1u0) [inline packages/frontend-v2/README.md:7]: should this be `yarn dev:v2`?

File: README.md (hunk 4 above)
Comment (cindy1u0) [inline README.md:3]: I am wondering if we can link the V2 README here for New Version, and link the old README below

File: .github/workflows/main.yml (hunk 3 above)
Comment (bugsalexander) [reply .github/workflows/main.yml:14]: the goal is to eventually upgrade everything to v16, i was hoping we could do scrapers to 16 during the rewrite

File: packages/frontend-v2/.eslintrc.js (hunk 5 above)
Comment (aryanshah701) [reply packages/frontend-v2/.eslintrc.js:2]: So the way I understand it, eslint cascades and merges eslint configs from the directory of the file all the way up to root. You can test this out by adding a unused var and you'll see the eslint rule "unused vars should be prefixed with an underscore" pop up and that rule is coming from the root eslint. I got this stuff from the [docs](https://eslint.org/docs/user-guide/configuring/configuration-files) under the `cascading and hierarchy` section but let me know if understood this wrong!

File: packages/frontend-v2/package.json (hunk 6 above)
Comment (aryanshah701) [reply packages/frontend-v2/package.json:12]: ooh nice thank you for catching this!!

File: packages/frontend/Dockerfile (hunk 9 above)
Comment (aryanshah701) [reply packages/frontend/Dockerfile:13]: yea lmao, probably going to kick the can for later cause we're moving to the new frontend anyways... we can nail down the infra of our new stack once we're done(maybe not even use docker?!)

File: packages/frontend-v2/README.md (hunk 10 above)
Comment (aryanshah701) [reply packages/frontend-v2/README.md:7]: yess thank you!!

File: README.md (hunk 4 above)
Comment (aryanshah701) [reply README.md:3]: oh as in have 2 seperate READMEs for V1 and V2?

Comment (aryanshah701) [reply README.md:3]: there's a lot of junk in our README actually (the whole available scripts) and also what even is the `GraduateNU - Backend` section, did we have a NodeJS backend at some point?

**Review by bugsalexander — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (cindy1u0) [reply README.md]: hmm i dont think so...

**Review by cindy1u0 — APPROVED** [review]
pogchamp

**Pull request merged**